backend/taxHandlers.ts

```diff
@@ -8,13 +8,70 @@ import {
   FetchTaxesPayload,
   TaxJarConfig,
   OrderPayload,
+  LinePayload,
+  DiscountPayload,
+  FetchTaxesLinePayload,
 } from "./types";
 
+const getDiscountForLine = (
+  line: LinePayload,
+  totalDiscount: number,
+  allLinesTotal: number
+) => {
+  if (totalDiscount === 0 || allLinesTotal == 0) {
+    return 0;
+  }
+  const lineTotalAmount = Number(line.total_amount);
+  const discountAmount = (lineTotalAmount / allLinesTotal) * totalDiscount;
+  if (discountAmount > lineTotalAmount) {
+    return lineTotalAmount;
+  }
+  return discountAmount;
+};
+
+const prepareLinesPayload = (
+  lines: Array<LinePayload>,
+  discounts: Array<DiscountPayload>
+): Array<FetchTaxesLinePayload> => {
+  const allLinesTotal = lines.reduce(
+    (total, current) => total + +current.total_amount,
```

Comment:
Since you're using `Number()` in other places to convert string to number I think it would be nice to stick to one way of conversion. The `+` trick isn't that explicit and readable 😬 

```suggestion
    (total, current) => total + Number(current.total_amount),
```